src/core/Table/filters/FilterWrapper.tsx

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import '@itwin/itwinui-css/css/table.css';
+import Popover from '../../utils/Popover';
+import { useTheme } from '../../utils/hooks/useTheme';
+import { ColumnInstance, HeaderGroup } from 'react-table';
+import { TableFilterValue } from './types';
+
+export type FilterWrapperProps<T extends Record<string, unknown>> = {
+  column: HeaderGroup<T>;
+  allColumns: ColumnInstance<T>[];
+  children: JSX.Element;
+  filterBody: React.ReactNode;
+  onShow: () => void;
+  onHide: () => void;
+};
+
+export const FilterWrapper = <T extends Record<string, unknown>>(
+  props: FilterWrapperProps<T>,
+) => {
+  const { column, children, filterBody, onShow, onHide, allColumns } = props;
+
+  useTheme();
+
+  const [isVisible, setIsVisible] = React.useState(false);
+  const close = React.useCallback(() => setIsVisible(false), []);
+
+  const setFilter = React.useCallback(
+    (filterValue: TableFilterValue<T> | undefined) => {
+      column.setFilter(filterValue);
+      close();
+    },
+    [close, column],
+  );
+
+  const clearFilter = React.useCallback(() => {
+    column.setFilter(undefined);
+    close();
+  }, [close, column]);
+
+  const placement = React.useMemo(() => {
+    const columnIndex = allColumns.findIndex((c) => c.id === column.id);
+    const columnsCount = allColumns.length;
+    if (columnIndex === 0) {
+      return 'bottom-start';
+    }
+    if (columnIndex === columnsCount - 1) {
+      return 'bottom-end';
+    }
+    return 'bottom';
```

Comment:
Is it really bad if it's always `bottom`?

Reply:
Hm, with popper maybe it is not a big issue as it positions itself to fit on screen:
![image](https://user-images.githubusercontent.com/36186912/117767872-34e1e980-b23a-11eb-84dd-130e26dfc13f.png)